core/domain/activity_validators.py

```diff
@@ -0,0 +1,90 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validators for activity models."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from constants import constants
+from core.domain import activity_domain
+from core.domain import base_model_validators
+from core.platform import models
+import feconf
+
+ERROR_CATEGORY_PROPERTY_FETCH_CHECK = 'fetch properties'
+
+(
+    collection_models, exp_models,) = (
+        models.Registry.import_models([
+            models.NAMES.collection, models.NAMES.exploration]))
+
+
+class ActivityReferencesModelValidator(
+        base_model_validators.BaseModelValidator):
+    """Class for validating ActivityReferencesModels."""
+
+    @classmethod
+    def _get_model_id_regex(cls, unused_item):
+        # Valid id: featured.
+        regex_string = '^(%s)$' % '|'.join(
+            feconf.ALL_ACTIVITY_REFERENCE_LIST_TYPES)
+        return regex_string
+
+    @classmethod
+    def _get_model_domain_object_instance(cls, item):
+        activity_references_list = []
+
+        try:
+            for reference in item.activity_references:
+                activity_references_list.append(
+                    activity_domain.ActivityReference(
+                        reference['type'], reference['id']))
+        except Exception as e:
+            cls._add_error(
+                ERROR_CATEGORY_PROPERTY_FETCH_CHECK,
+                'Entity id %s: Entity properties cannot be fetched completely '
```

Comment:
Should the message have info about this appeared from `_get_model_domain_object_instance` function and not the other? (Asking this as other method has the same error log.) Is this international to avoid kinda duplicate errors? not sure!

Reply:
De-duplication is already done when validation job is run. We just have repeated messages to avoid any confusion around why we are generating error at one place and not other.